src/KeyedFrames.jl

```diff
@@ -174,17 +174,25 @@ function DataFrames.deleterows!(kf::KeyedFrame, ind)
     return kf
 end
 
-@deprecate deletecols!(kf::KeyedFrame, ind) select!(kf, ind)
+@deprecate deletecols!(kf::KeyedFrame, ind) select!(kf, Not(ind))
 
-DataFrames.select!(kf::KeyedFrame, ind::Union{Integer, Symbol}) = select!(kf, [ind])
-DataFrames.select!(kf::KeyedFrame, ind::Vector{<:Integer}) = select!(kf, names(kf)[ind])
+DataFrames.select!(kf::KeyedFrame, ind::Integer) = select!(kf, [ind])
+DataFrames.select!(kf::KeyedFrame, ind::Symbol) = select!(kf, index(kf)[ind])
 
-function DataFrames.select!(kf::KeyedFrame, ind::Vector{<:Symbol})
-    select!(frame(kf), Not(ind))
-    filter!(x -> !in(x, ind), keys(kf))
+function DataFrames.select!(kf::KeyedFrame, ind)
+    select!(frame(kf), ind)
+    new_keys = names(frame(kf))
+    filter!(x -> in(x, new_keys), keys(kf))
     return kf
 end
 
+function DataFrames.select(kf::KeyedFrame, ind)
+    new_df = select(frame(kf), ind)
+    df_names = names(new_df)
+    new_keys = filter(x -> in(x, df_names), keys(kf))
+    return KeyedFrame(new_df, new_keys)
+end
```

Comment:
Based upon what I see in the `DataFrames` internals we just need to implement:
```julia
select!(df::DataFrame, inds::AbstractVector{Int})
select(df::DataFrame, inds::AbstractVector{Int}; copycols::Bool=true)
```
as the other methods are fairly generic:
```julia
select!(df::DataFrame, c::Int) = select!(df, [c])
select!(df::DataFrame, c::Any) = select!(df, index(df)[c])

select(df::DataFrame, c::Int; copycols::Bool=true) = select(df, [c], copycols=copycols)
select(df::DataFrame, c::Any; copycols::Bool=true) = select(df, index(df)[c], copycols=copycols)
```

Reply:
Oh wait. They made those methods `DataFrame` specific so we'll need to extend them for `KeyedFrame`s but we can keep the same definition otherwise

Reply:
We can just define

```julia

function DataFrames.select!(kf::KeyedFrame, inds)
    select!(frame(kf), inds)
    new_keys = names(frame(kf))
    filter!(x -> in(x, new_keys), keys(kf))
    return kf
end

function DataFrames.select(kf::KeyedFrame, inds; copycols::Bool=true)
    new_df = select(frame(kf), inds; copycols=copycols)
    df_names = names(new_df)
    new_keys = filter(x -> in(x, df_names), keys(kf))
    return KeyedFrame(new_df, new_keys)
end
```

And the other methods should be covered as they call the DataFrames methods

Reply:
Ya, just do that. I would just do this in both:
```julia
new_keys = filter!(in(df_names), keys(kf))
```

Reply:
I'll have to use `filter` in the second case as that modifies the keys for the passed-in keyedframe and not just the copy